src/Xamarin.Android.Build.Tasks/Utilities/Files.cs

```diff
@@ -312,28 +312,22 @@ public static ZipArchive ReadZipFile (string filename, bool strictConsistencyChe
 				}
 				var outfile = Path.GetFullPath (Path.Combine (destination, fullName));
 				files.Add (outfile);
-				var dt = File.Exists (outfile) ? File.GetLastWriteTimeUtc (outfile) : DateTime.MinValue;
-				if (forceUpdate || entry.ModificationTime > dt) {
-					try {
-						entry.Extract (destination, fullName, FileMode.Create);
-						MonoAndroidHelper.SetWriteable (outfile);
-						var utcNow = DateTime.UtcNow;
-						File.SetLastWriteTimeUtc (outfile, utcNow);
-						File.SetLastAccessTimeUtc (outfile, utcNow);
-					} catch (PathTooLongException) {
-						throw new PathTooLongException ($"Could not extract \"{fullName}\" to \"{outfile}\". Path is too long.");
-					}
-					updated = true;
+				try {
+					entry.Extract (destination, fullName, FileMode.Create);
+					MonoAndroidHelper.SetWriteable (outfile);
+				} catch (PathTooLongException) {
+					throw new PathTooLongException ($"Could not extract \"{fullName}\" to \"{outfile}\". Path is too long.");
 				}
+				updated = true;
 			}
 			foreach (var file in Directory.GetFiles (destination, "*.*", SearchOption.AllDirectories)) {
 				var outfile = Path.GetFullPath (file);
 				if (outfile.Contains ("/__MACOSX/") ||
-				    		outfile.EndsWith ("__AndroidLibraryProjects__.zip", StringComparison.OrdinalIgnoreCase) ||
+						outfile.EndsWith ("__AndroidLibraryProjects__.zip", StringComparison.OrdinalIgnoreCase) ||
 						outfile.EndsWith ("/__MACOSX", StringComparison.OrdinalIgnoreCase) ||
 						outfile.EndsWith ("/.DS_Store", StringComparison.OrdinalIgnoreCase))
 					continue;
-				if (!files.Contains (outfile) && !(deleteCallback?.Invoke (outfile) ?? true)) {
+				if (!files.Contains (outfile) && (deleteCallback?.Invoke (outfile) ?? true)) {
```

Comment:
@dellis1972 this was the line here I thought the logic was backwards... let me know what you think.

Reply:
It's not deleting the entire directory anymore, since removing this `!` things worked correctly.

Do you think it was just reversed?